packages/schema/src/types/module.ts

```diff
@@ -26,19 +26,32 @@ export interface ModuleMeta {
 /** The options received.  */
 export type ModuleOptions = Record<string, any>
 
+/** Optional result for nuxt modules */
+export interface ModuleReturn {
+  /**
+   * Timing information for the initial setup
+   */
+  timings?: {
+    /** Total time took for module setup in ms */
+    setup?: number
+    [key: string]: number | undefined
+  }
+}
+
+type Awaitable<T> = T | Promise<T>
+export type _ModuleReturn = Awaitable<void | false | ModuleReturn>
```

Comment:
Is there a reason we are exporting this internal type?

Reply:
I saw a recommendation somewhere before to always export public interfaces. ModuleSetupReturn seems good idea.